src/python_bluesky_taskgraph/core/decision_engine.py

```diff
@@ -60,6 +57,7 @@ def __init__(
         self._should_stop_at_end_of_next_run: bool = False
         self._error_tasks: Dict[str, int] = {}
         self._recovered_tasks: Set[str] = set()
+        self._logger = logging.getLogger(self.__class__.__name__)
```

Comment:
Can have loggers at the module level rather than the class level for less boilerplate, so at the top of the file:

```python
LOGGER = logging.getLogger(__name__)
```

Reply:
Depends on whether there'd ever be more than one DecisionEngine, [of different types] that we might want to distinguish logging for. I think with the same argument as having a single RunEngine, that I oppose that so I'm happy to make this change: and it simplifies the logging exception handling.